halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/model/v1/node/Validatable.java

```diff
@@ -14,18 +14,13 @@
  * limitations under the License.
  */
 
-package com.netflix.spinnaker.halyard.config.validate.v1;
+package com.netflix.spinnaker.halyard.config.model.v1.node;
 
-import java.lang.annotation.ElementType;
-import java.lang.annotation.Retention;
-import java.lang.annotation.RetentionPolicy;
-import java.lang.annotation.Target;
+import com.netflix.spinnaker.halyard.config.model.v1.problem.ProblemSetBuilder;
 
 /**
- * An annotation to validate a specific field.
+ * Anything implementing this interface is validatable, meaning after we validate it, we get a non-null list of HalconfigProblems
  */
-@Retention(RetentionPolicy.RUNTIME)
-@Target(ElementType.FIELD)
-public @interface ValidateField {
-  Class<? extends Validator<?>>[] validators();
+public interface Validatable {
+  void accept(ProblemSetBuilder psBuilder, Validator v);
```

Comment:
There is only one way - this is standard in the visitor pattern because as we iterate over `Node`s, we want to to ensure that the Validator is called on the right `Node` subclass, so we have the `Node` pass `this` into the `Validator` which will be the concrete subclass we want to validate.